### ISSUE TITLE
Fio.open creates files with 000 UNIX mode

### ISSUE THREAD

**Issue by mejedi** [tarantool/tarantool#1211]
It would be great if it was possible to pass desired mode to `fio.open`. Use sane defaults if not passed.
```
tarantool> fio=require'fio'
---
...

tarantool> fio.open('42', {'O_WRONLY','O_CREAT'})
---
- fh: 6
...

$ stat 42
16777220 10664749 ---------- 1 n.zavaritsky staff 0 0 "Dec 15 18:58:03 2015" "Dec 15 18:58:03 2015" "Dec 15 18:58:03 2015" "Dec 15 18:58:03 2015" 4096 0 0 42
```

**Comment by kostja** [comment]
Th.open should behave just like plain UNIX open: if the third paramete is omitted, the file is created in 777 & umask mode. This should be a one line fix, please just close it.

**Comment by mejedi** [comment]
>Th.open should behave just like plain UNIX open: if the third paramete is omitted, the file is created in 777 & umask mode.
I believe UNIX `open` will just lift random garbage from stack if the third param was omitted.

**Comment by rtsisyk** [comment]
The same issue for mkdir()

**Comment by mejedi** [comment]
A few notes:
 * `fio.open`, `fio.mkdir` and `fio.chmod` gets a mode, but only the first one implements symbolic modes, e.g. `fio.open('pathname', { 'O_RDWR' }, { 'S_IXUSR', 'S_IWUSR' } )`;

 * the later style feels awkward, probably we can design a better one? draw inspiration from other Lua libraries;

 * `mode` parameter in `fio.open` and `fio.mkdir` not documented.

**Comment by kostja** [comment]
* Nick Zavaritsky <notifications@github.com> [16/03/23 21:32]:
>  * `fio.open`, `fio.mkdir` and `fio.chmod` gets a mode, but only
>  the first one implements symbolic modes, e.g.
>  `fio.open('pathname', { 'O_RDWR' }, { 'S_IXUSR', 'S_IWUSR' }
>  )`;

Maybe, when someone wants this feature.

>  * the later style feels awkward, probably we can design a
>  better one? draw inspiration from other Lua libraries;

Ditto. Upon user request.

>  * `mode` parameter in `fio.open` and `fio.mkdir` not documented.

Please file a separate documentation request.

-- 
Konstantin Osipov, Moscow, Russia, +7 903 626 22 32
http://tarantool.org - www.twitter.com/kostja_osipov

**Comment by mejedi** [comment]
Filed #1404

**Comment by Totktonada** [comment]
It was fixed in 1.6.8-593-gd92e61bb4.
